community/kernel/src/main/java/org/neo4j/kernel/DatabaseAvailability.java

```diff
@@ -63,18 +76,18 @@ public void stop()
     {
         // Database is no longer available for use
         // Deny beginning new transactions
-        availabilityGuard.deny(this);
+        availabilityGuard.deny( this );
 
         // Await transactions stopped
-        awaitNoTransactionsOr( 10_000  /* ms */);
+        awaitTransactionsClosedWithinTimeout();
     }
 
-    private void awaitNoTransactionsOr( int orUntilDeadline )
+    private void awaitTransactionsClosedWithinTimeout()
     {
-        long deadline = SYSTEM_CLOCK.currentTimeMillis() + orUntilDeadline;
-        while ( transactionMonitor.getNumberOfActiveTransactions() > 0 && SYSTEM_CLOCK.currentTimeMillis() < deadline)
+        long deadline = SYSTEM_CLOCK.currentTimeMillis() + awaitActiveTransactionDeadlineMillis;
+        while ( transactionMonitor.getNumberOfActiveTransactions() > 0 && SYSTEM_CLOCK.currentTimeMillis() < deadline )
         {
-            Thread.yield();
+            parkNanos( MILLISECONDS.toNanos( 10 ) );
```

Comment:
To me the main difference is that we add extra cruft to the code which would not add anything. If you really want it I'd like to add some sort of sleep utility w/ regards to interruption, so that the code wouldn't need to be contaminated with that here.